File: benchmark/2026-09-17-7677581(latest)/standard/rubrics/probes/CONC-1/kotlin/conc1.kt
// CONC-1, Kotlin/JVM. Worker mechanism: kotlin.concurrent.thread (Kotlin standard library).
import kotlin.concurrent.thread

const val N: Long = 500000000L
const val CHUNKS: Int = 4
const val SPAN: Long = N / CHUNKS

val partial = DoubleArray(CHUNKS)

fun chunkSum(c: Int): Double {
    var s = 0.0
    val start = c.toLong() * SPAN
    val end = start + SPAN
    var i = start
    while (i < end) { val x = Math.sin(i.toDouble()); s += x * x; i++ }
    return s
}

fun main(args: Array<String>) {
    val w = if (args.isNotEmpty()) args[0].toInt() else 1
    val ths = (0 until w).map { wi ->
        thread(start = true) {
            for (c in 0 until CHUNKS) if (c % w == wi) partial[c] = chunkSum(c)
        }
    }
    ths.forEach { it.join() }
    var total = 0.0
    for (c in 0 until CHUNKS) total = total + partial[c]
    println(String.format("workers=%d result=%.10f", w, total))
}

File: benchmark/2026-09-17-7677581(latest)/standard/rubrics/probes/CONC-1/go/conc1.go
// CONC-1, Go. Worker mechanism: goroutines with sync.WaitGroup (standard library).
package main

import (
	"fmt"
	"math"
	"os"
	"strconv"
	"sync"
)

const N int64 = 500000000
const CHUNKS int = 4
const SPAN int64 = N / int64(CHUNKS)

func chunkSum(c int) float64 {
	s := 0.0
	start := int64(c) * SPAN
	end := start + SPAN
	for i := start; i < end; i++ {
		x := math.Sin(float64(i))
		s += x * x
	}
	return s
}

func main() {
	W := 1
	if len(os.Args) > 1 {
		W, _ = strconv.Atoi(os.Args[1])
	}
	partial := make([]float64, CHUNKS)
	var wg sync.WaitGroup
	for w := 0; w < W; w++ {
		wg.Add(1)
		go func(w int) {
			defer wg.Done()
			for c := 0; c < CHUNKS; c++ {
				if c%W == w {
					partial[c] = chunkSum(c)
				}
			}
		}(w)
	}
	wg.Wait()
	total := 0.0
	for c := 0; c < CHUNKS; c++ {
		total = total + partial[c]
	}
	fmt.Printf("workers=%d result=%.10f\n", W, total)
}
